File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter05/CallContextRemoting/MathLibrary/CallContextData.vb
Imports System.Runtime.Remoting.Messaging 'For ILogicalThreadAffinative

<Serializable()> _
Public Class CallContextData
   Implements ILogicalThreadAffinative
   
   Private mData As Object
   
   Public Sub New(data As Object)
      mData = data
   End Sub

   Public Readonly Property Data() As Object
      Get 
         Return mData
      End Get
   End Property
End Class

File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter05/CallContextAsync/MathLibrary/CallContextData.cs
using System;
using System.Runtime.Remoting.Messaging; // Call Context stuff

namespace MathLibrary
{
[Serializable]
public class CallContextData : ILogicalThreadAffinative
{
   private object mData;
	public CallContextData(object data)
	{
		mData = data;
	}

   public object Data
   {
      get { return mData; }
   }
}
}
